src/compas/artists/primitiveartist.py

```diff
@@ -50,10 +50,16 @@ def primitive(self, primitive):
     @property
     def color(self):
         if not self._color:
-            self._color = self.default_color
+            self.color = self.default_color
         return self._color
 
     @color.setter
-    def color(self, color):
-        if is_color_rgb(color):
-            self._color = color
+    def color(self, c):
+        if not c:
```

Comment:
very small thing, maybe `if c is not None`, cuz this is going to raise error when c is an np array